src/index.js

```diff
@@ -621,22 +622,23 @@ class Manager {
       return this._spawn(binPath, sanityCheck.args);
     })
     .then((output) => {
-      const haystack = output.stdout + output.stderr;
+      this._logger.debug(`Sanity check output: ${output.stdout + output.stderr}`);
 
-      this._logger.debug(`Sanity check output: ${haystack}`);
-
-      const needles = sanityCheck.output || [];
+      try {
+        var version = output.stdout.match(/\d+\.\d+\.\d+/)[0];
+      } catch (err) {
+        throw new Error(`Unable to find semantic version >= "${client.version}" in "${binPath} ${sanityCheck.args}"'s' output`);
+      }
 
-      for (let needle of needles) {
-        if (0 > haystack.indexOf(needle)) {
-          throw new Error(`Unable to find "${needle}" in ${client.id} output`);
-        }
+      if (version && semver.lt(version, client.version)) {
```

Comment:
We should not be hardcoding this semver check. We should instead refactor the sanity check configuration system such that such logic can be easily represented. I would recommend adding a `versionCommand` config parameter which specifies the CLI arguments to use in order to obtain binary version. Then we can do this semver check.